### PR TITLE
Adding a new schema for taskcluster.yml v1

### PR DESCRIPTION
**Pull request by owlishDeveloper**
As a part of the JSON-e-fication project, I think I need to add a new schema to validate new `taskcluster.yml`s v1. After finishing (or preliminary finishing) the documentation, I began implementing. TDD is my technique of choice here, but it seems to me I need the schema to be _up there_ somewhere (I am having trouble reading it from a nearby directory). The tests, I imagine, should validate the YAMLs.

My understanding is, in order to get a schema _up there_, I need to add a yaml file to that effect to the `schemas` dir.

So, the new schema for your consideration!

### CODE REVIEW

File: schemas/taskcluster-github-config.v1.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          detailed definition of the options.
+      type: string
+      enum:
+        - public
+        - collaborators
+  tasks:
+    type:           array
+    default:        []
+    items:
+        title:              "Task Definition"
```
Comment (imbstack) [inline schemas/taskcluster-github-config.v1.yml]: I think these will be better called something like "task definition template" maybe? They don't need to be task definitions until after this is rendered. Similarly with the `created` and `deadline` fields. Probably don't need to be asserted by this schema? I might be missing something though!

**Review by imbstack — APPROVED** [review]
Little nitpick I have but otherwise looks good!

File: schemas/taskcluster-github-config.v1.yml (hunk 1 above)
Comment (owlishDeveloper) [reply schemas/taskcluster-github-config.v1.yml]: I gave it a thunk, and it seems to me we don't have to assert anything. `created` comes easily from context, we can have some default deadline... Why did we assert them in v0?

Comment (djmitche) [reply schemas/taskcluster-github-config.v1.yml]: In hooks, we call these a "task template", so we could do the same here.

We also had to remove asserting created/deadline/expires in hooks.  The template can create all three easily enough with `$fromNow`.

File: schemas/taskcluster-github-config.v1.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+$schema:            http://json-schema.org/draft-06/schema#
+title:              ".taskcluster.yml format"
+description: |
+ Description of a taskclusterrc.yml file v1, which may be used to generate a taskgraph
```
Comment (djmitche) [inline schemas/taskcluster-github-config.v1.yml]: I don't know if this really matters, but that filename is wrong :)

File: schemas/taskcluster-github-config.v1.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ and tasks.
+type:               object
+properties:
+  version:
+    description:    ".taskcluster.yml version"
```
Comment (djmitche) [inline schemas/taskcluster-github-config.v1.yml]: "Version of the format of this file; must be 1"

File: schemas/taskcluster-github-config.v1.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      enum:
+        - public
+        - collaborators
+  tasks:
+    type:           array
+    default:        []
+    items:
+        title:              "Task definition template"
+        description: |
+          Definition of a task that can be scheduled
```
Comment (djmitche) [inline schemas/taskcluster-github-config.v1.yml]: It's worth mentioning this is rendered with JSON-e, and linking to the main document.

File: schemas/taskcluster-github-config.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 $schema:            http://json-schema.org/draft-06/schema#
 title:              ".taskcluster.yml format"
 description: |
- Description of a taskclusterrc.yml file, which may be used to generate a taskgraph
+ Description of a taskclusterrc.yml file v0, which may be used to generate a taskgraph
```
Comment (djmitche) [inline schemas/taskcluster-github-config.yml]: Heh, this filename is wrong too :)

**Review by djmitche — APPROVED** [review]
A few more minor nitpicks, but ++

File: schemas/taskcluster-github-config.v1.yml (hunk 4 above)
Comment (owlishDeveloper) [reply schemas/taskcluster-github-config.v1.yml]: I'll mention JSONe, but will add the link in the documentation PR

**Comment by owlishDeveloper** [comment]
This can be merged, right? 🙏

**Comment by imbstack** [comment]
tada!

**Pull request merged**